src/firmware/boards/legacy_dongle_stm32f4/normal.c

```diff
@@ -973,6 +960,12 @@ static void dongle_status_task(void *UNUSED(param))
                 (second_dongle ? 0x08U : 0x00U);
             state |= ((transmit_queue_full ? 0x10U : 0x00U) |
                       (receive_queue_full ? 0x20U : 0x00U));
+            state |=
+                __atomic_exchange_n(&attempted_to_send_packet_larger_then_max_packet_size,
+                                    false, __ATOMIC_RELAXED)
```

Comment:
Can we get a comment explaining what's going on here. It's not easy to follow.

Reply:
So I think this could get a bit out of hand. If I comment this then I should really comment the whole thing, but even then this should really all just get rewritten. I'm going to say we should just leave it for now since it's at least consistent with what's here.